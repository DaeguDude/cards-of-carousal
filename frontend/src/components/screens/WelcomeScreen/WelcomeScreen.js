import React from 'react';
import styled from 'styled-components';
import Header from '../../Header/Header';
import Button from '../../Buttons/Button';

const WelcomeScreenWrapper = styled.div`
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  display: flex;
  flex-direction: column;
  background-color: var(--primary-background-color);

  /*Header*/
  .header-container {
    display: flex;
    height: 190px;
    flex-direction: column;
    justify-content: flex-end;
  }

  .welcome-to {
    margin-top: 0px;
    line-height: 2rem;
    margin-bottom: -0.9rem;
  }

  .CoC {
    font-size: 6rem;
    margin-bottom: -1rem;
    vertical-align: bottom;
    line-height: 6rem;
  }

  /*Main Body*/
  .main-body-container {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    margin: auto;
  }

  .definition-container {
    display: flex;
    flex-direction: column;
    font-size: 3.2rem;
    font-weight: 700;
    transform: rotate(-5deg);
    margin-bottom: 9rem;
    line-height: 3.3rem;
  }

  .definition {
    font-size: 2.75rem;
    font-weight: 400;
  }

  .button-container {
    display: flex;
    flex-direction: row;
    align-items: center;
  }

  .join-btn {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 16rem;
    height: 3rem;
    font-size: 2.5rem;
    font-weight: 900;
  }

  .OR {
    display: flex;
    align-items: center;
    justify-content: center;
    background-color: var(--secondary-background-color);
    color: var(--secondary-text-color);
    border-radius: 50%;
    padding: 1rem;
    width: 4.5rem;
    height: 4.5rem;
    margin-left: 5.8rem;
    margin-right: 5.8rem;
    font-size: 2.25rem;
    font-weight: 700;
  }

  /*Footer*/
  .footer-container {
    display: flex;
    flex-direction: column;
    align-items: flex-end;
    justify-content: flex-end;
    margin-top: auto;
    height: 140px;
    background-color: var(--secondary-background-color);
    color: var(--secondary-text-color);
    font-size: 2rem;
    font-weight: 400;
    padding-right: 20px;
  }

  .credit {
    margin-top: 5rem 3rem;
  }

  /*small size hosting screen */
  @media (max-width: 970px) {
    .footer-container {
      text-align: center;
      font-size: 1.5rem;
    }

    .definition-container {
      margin-bottom: 5rem;
    }

    .CoC {
      font-size: 4rem;
      margin-bottom: -1.3rem;
      line-height: 5rem;
    }
  }

  /*largest desktop player mode */
  @media (max-width: 785px) {
    .header-container {
      height: 180px;
    }

    .CoC {
      font-size: 4rem;
      margin-bottom: -0.95rem;
      line-height: 4.5rem;
    }

    .definition-container {
      display: none;
      margin-bottom: 3rem;
    }

    .join-btn {
      width: 14rem;
    }

    .host-btn {
      display: none;
    }

    .OR {
      display: none;
    }

    .footer-container {
      align-items: center;
      justify-content: center;
      height: 120px;
      background-color: transparent;
      color: var(--accent-text-color);
      text-align: center;
      font-size: 1.2rem;
      padding-bottom: 2rem;
    }

    .credit {
      line-height: 1.5rem;
      margin: 0.5rem;
      padding-left: 140px;
      padding-right: 140px;
    }
  }

  /* Keeps footer content formatted correctly as player only screen shrinks on desktop*/
  @media (max-width: 645px) {
    .welcome-to {
      display: none;
    }

    .credit {
      line-height: 1.5rem;
      margin: 0.5rem;
      padding-left: 70px;
      padding-right: 70px;
    }
  }

  /* modern larger mobile devices */
  @media (max-width: 500px) {
    .credit {
      padding-left: 20px;
      padding-right: 20px;
    }
  }

  /*iphone 5*/
  @media (max-width: 320px) {
    .header-container {
      height: 130px;
    }

    .CoC {
      font-size: 3rem;
      margin-bottom: -0.75rem;
      vertical-align: bottom;
      line-height: 3.5rem;
    }

    .footer-container {
      font-size: 1rem;
    }
  }
`;

function WelcomeScreen() {
  return (
    <WelcomeScreenWrapper className="primary-background">
      <Header className="header-container">
        <h1 className="welcome-to">WELCOME TO</h1>
        <h2 className="CoC">CARDS OF CAROUSAL</h2>
      </Header>
      <main className="main-body-container">
        <div className="definition-container">
          <div className="word">
            CAROUSAL <span className="definition">(n)</span>
          </div>
          <div className="definition">
            a wild, drunken party or celebration : a drunken revel
          </div>
        </div>
        <div className="button-container">
          <Button type="button" isActive className="join-btn">
            <p> JOIN</p>
          </Button>
          <div className="OR">OR</div>
          <Button type="button" isActive className=" join-btn host-btn">
            <p>HOST</p>
          </Button>
        </div>
      </main>
      <footer className="footer-container">
        <div className="credit">
          Card content thanks to: https://cardsagainsthumanity.com
        </div>
        <div className="credit">
          Made by Odin students, with love: https://www.theodinproject.com
        </div>
      </footer>
    </WelcomeScreenWrapper>
  );
}

export default WelcomeScreen;
