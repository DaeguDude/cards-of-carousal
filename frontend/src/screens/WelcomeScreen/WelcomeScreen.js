import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Button from '../../components/Buttons/Button';
import HeaderFooterLayout from '../../layouts/HeaderFooterLayout/HeaderFooterLayout';

const propTypes = {
  handleJoinClick: PropTypes.func.isRequired,
  handleHostClick: PropTypes.func.isRequired,
};

const WelcomeScreenWrapper = styled.div`
  display: flex;
  flex-direction: column;

  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  height: 100%;
  background-color: var(--primary-background-color);

  main {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    margin: auto;
  }

  .definition-container {
    display: flex;
    flex-direction: column;

    font-size: 5.2rem;
    font-weight: 700;
    line-height: 4.2rem;

    transform: rotate(-5deg);
    margin-bottom: 190px;
  }

  .definition {
    font-size: 4rem;
    font-weight: 400;
  }

  .definition-container p {
    display: inline-block;
  }

  .button-container {
    display: flex;
    flex-direction: row;
    align-items: center;
  }

  .button {
    display: flex;
    justify-content: center;
    align-items: center;

    font-size: 6rem;
    font-weight: 900;

    width: 317px;
    height: 82px;

    font-size: 3.6rem;
  }

  .OR {
    display: flex;
    align-items: center;
    justify-content: center;

    background-color: var(--secondary-background-color);
    color: var(--secondary-text-color);

    border-radius: 50%;
    padding: 16px;

    font-size: 2.3rem;
    font-weight: 700;

    width: 82px;
    height: 82px;
    margin-left: 93px;
    margin-right: 93px;
  }

  @media (max-width: 1920px) {
    .definition-container {
      font-size: 4.5rem;
      line-height: 3.7rem;

      margin-bottom: 150px;
    }

    .definition {
      font-size: 3.5rem;
    }

    .button {
      width: 290px;
      height: 74px;

      font-size: 3.2rem;
    }

    .OR {
      width: 74px;
      height: 74px;
      margin-left: 93px;
      margin-right: 93px;

      font-size: 2rem;
    }
  }

  @media (max-width: 1440px) {
    .definition-container {
      font-size: 3.2rem;
      line-height: 2.6rem;

      margin-bottom: 90px;
    }

    .definition {
      font-size: 2.5rem;
    }

    .button {
      width: 256px;
      height: 66px;

      font-size: 2.5rem;
    }

    .OR {
      width: 66px;
      height: 66px;
      margin-left: 93px;
      margin-right: 93px;

      font-size: 1.8rem;
    }
  }

  @media (max-width: 950px) {
    .definition-container {
      font-size: 2.4rem;
      line-height: 1.8rem;
    }

    .definition {
      font-size: 1.85rem;
    }

    .button {
      width: 200px;
      height: 51px;

      font-size: 2rem;
    }

    .OR {
      width: 51px;
      height: 51px;
      margin-left: 70px;
      margin-right: 70px;

      font-size: 1.4rem;
    }
  }

  /*if less than 700px, or on mobile landscape, do this.... */
  @media (max-width: 700px), (max-width: 900px) and (max-height: 500px) {
    .definition-container {
      display: none;
    }

    .host-button {
      display: none;
    }

    .OR {
      display: none;
    }
  }

  /* Screens above 1440p */
  @media (min-width: 2561px) {
    .definition-container {
      font-size: 6.5rem;
      line-height: 4.8rem;

      margin-bottom: 300px;
    }

    .definition {
      font-size: 5rem;
    }

    .button {
      width: 490px;
      height: 126px;

      font-size: 5.2rem;
    }

    .OR {
      width: 126px;
      height: 126px;
      margin-left: 93px;
      margin-right: 93px;

      font-size: 3.8rem;
    }
  }
`;

function WelcomeScreen({ handleJoinClick, handleHostClick }) {
  return (
    <WelcomeScreenWrapper className="primary-background">
      <HeaderFooterLayout isWelcoming>
        <main>
          <div className="definition-container">
            <p className="word">
              CAROUSAL <span className="definition">(n)</span>
            </p>
            <p className="definition">
              a wild, drunken party or celebration : a drunken revel
            </p>
          </div>

          <div className="button-container">
            <Button
              type="button"
              isActive
              className="button"
              onClick={handleJoinClick}
            >
              <p>JOIN</p>
            </Button>

            <div className="OR">OR</div>

            <Button
              type="button"
              isActive
              className="button host-button"
              onClick={handleHostClick}
            >
              <p>HOST</p>
            </Button>
          </div>
        </main>
      </HeaderFooterLayout>
    </WelcomeScreenWrapper>
  );
}

export default WelcomeScreen;

WelcomeScreen.propTypes = propTypes;
