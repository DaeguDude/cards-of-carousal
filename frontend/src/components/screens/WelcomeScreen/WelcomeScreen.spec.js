import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react';

import WelcomeScreen from './WelcomeScreen';

describe('WelcomeScreen', () => {
  describe('render', () => {
    it('renders correctly', () => {
      const tree = renderer
        .create(
          <WelcomeScreen
            handleJoinClick={() => ''}
            handleHostClick={() => ''}
          />,
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('events', () => {
    it('fires callback when "join" button is clicked', () => {
      const mockJoinBtnCallback = jest.fn();
      const { getByText } = render(
        <WelcomeScreen
          handleJoinClick={mockJoinBtnCallback}
          handleHostClick={() => ''}
        />,
      );

      fireEvent.click(getByText('JOIN'));
      expect(mockJoinBtnCallback).toHaveBeenCalledTimes(1);
    });

    it('fires callback when "host" button is clicked', () => {
      const mockHostBtnCallback = jest.fn();
      const { getByText } = render(
        <WelcomeScreen
          handleJoinClick={() => ''}
          handleHostClick={mockHostBtnCallback}
        />,
      );

      fireEvent.click(getByText('HOST'));
      expect(mockHostBtnCallback).toHaveBeenCalledTimes(1);
    });
  });
});
