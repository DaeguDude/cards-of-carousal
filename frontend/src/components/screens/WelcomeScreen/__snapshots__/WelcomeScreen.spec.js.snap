// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeScreen render renders correctly 1`] = `
.c1 {
  width: 100%;
  background-color: var(--secondary-background-color);
  color: var(--secondary-text-color);
}

.c2 {
  display: block;
  border: 2px solid var(--primary-text-color);
  font-size: 12px;
  font-weight: bold;
  padding: 0.7em;
  background: var(--primary-background-color);
  color: var(--primary-text-color);
}

.c2:hover,
.c2:active,
.c2:focus {
  background: var(--accent-background-color);
  cursor: pointer;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  height: 200px;
  font-size: 3.5rem;
  font-weight: 400;
  padding-right: 16px;
  padding-bottom: 10px;
  background-color: var(--secondary-background-color);
  color: var(--secondary-text-color);
}

.c3 p {
  text-align: right;
  position: relative;
}

.c3 a {
  color: var(--secondary-text-color);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  height: 100%;
  background-color: var(--primary-background-color);
}

.c0 .header {
  height: 190px;
}

.c0 .header-text {
  position: relative;
  bottom: -122px;
}

.c0 .welcome-to {
  font-size: 2rem;
  line-height: 0.2rem;
  font-weight: 500;
  margin-top: 0px;
  margin-bottom: -14px;
}

.c0 .CoC {
  font-size: 5rem;
  line-height: 6.8rem;
  font-weight: 700;
  margin-bottom: -5px;
  vertical-align: bottom;
}

.c0 main {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: auto;
}

.c0 .definition-container {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 3rem;
  font-weight: 700;
  line-height: 2rem;
  -webkit-transform: rotate(-5deg);
  -ms-transform: rotate(-5deg);
  transform: rotate(-5deg);
  margin-bottom: 90px;
}

.c0 .definition {
  font-size: 2rem;
  font-weight: 400;
}

.c0 .definition-container p {
  display: inline-block;
}

.c0 .button-container {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .button {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 6rem;
  font-weight: 900;
  width: 256px;
  height: 48px;
  font-size: 2.5rem;
}

.c0 .OR {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: var(--secondary-background-color);
  color: var(--secondary-text-color);
  border-radius: 50%;
  padding: 16px;
  font-size: 2.25rem;
  font-weight: 700;
  width: 72px;
  height: 72px;
  margin-left: 93px;
  margin-right: 93px;
}

@media (max-width:2000px) {
  .c3 {
    height: 140px;
    font-size: 2rem;
  }
}

@media (max-width:970px) {
  .c3 {
    text-align: center;
    font-size: 1.5rem;
  }
}

@media (max-width:785px) {
  .c3 {
    height: 120px;
    background-color: transparent;
  }

  .c3 p {
    text-align: center;
    font-size: 1.2rem;
    padding-bottom: 32px;
    color: var(--accent-text-color);
  }

  .c3 a {
    color: var(--accent-text-color);
    display: block;
  }
}

@media (max-width:320px) {
  .c3 {
    font-size: 1rem;
    padding-right: 0px;
  }
}

@media (max-width:820px) {
  .c0 .header-text {
    bottom: -134px;
  }

  .c0 .welcome-to {
    font-size: 1.5rem;
  }

  .c0 .CoC {
    font-size: 4rem;
    line-height: 6rem;
  }

  .c0 .definition-container {
    font-size: 2.2rem;
    line-height: 1.8rem;
    margin-bottom: 90px;
  }

  .c0 .definition {
    font-size: 1.7rem;
  }

  .c0 .button {
    width: 200px;
    height: 48px;
    font-size: 2.5rem;
  }

  .c0 .OR {
    width: 60px;
    height: 60px;
    margin-left: 70px;
    margin-right: 70px;
    font-size: 2rem;
  }
}

@media (max-width:645px) {
  .c0 .header-text {
    bottom: -85px;
  }

  .c0 .definition-container {
    display: none;
  }

  .c0 .host-button {
    display: none;
  }

  .c0 .OR {
    display: none;
  }

  .c0 .welcome-to {
    display: none;
  }

  .c0 .CoC {
    line-height: 3.5rem;
  }
}

@media (max-width:320px) {
  .c0 .header {
    height: 130px;
  }

  .c0 .header-text {
    bottom: -30px;
  }

  .c0 .CoC {
    font-size: 3rem;
    margin-bottom: 12px;
  }
}

@media (min-width:2000px),(min-height:1500px) {
  .c0 .header {
    height: 400px;
  }

  .c0 .header-text {
    bottom: -260px;
  }

  .c0 .welcome-to {
    font-size: 3rem;
    line-height: 3.8rem;
  }

  .c0 .CoC {
    font-size: 8rem;
    line-height: 6.5rem;
  }

  .c0 .definition-container {
    font-size: 4.5rem;
    line-height: 3.5rem;
    margin-bottom: 250px;
  }

  .c0 .definition {
    font-size: 3.5rem;
  }

  .c0 .OR {
    font-size: 2.5rem;
    width: 100px;
    height: 100px;
    margin-left: 100px;
    margin-right: 100px;
  }

  .c0 .button {
    width: 500px;
    height: 100px;
    border: solid black 4px;
    font-size: 3rem;
  }
}

@media (min-width:2450px) {
  .c0 .header {
    height: 400px;
  }

  .c0 .header-text {
    bottom: -260px;
  }

  .c0 .welcome-to {
    font-size: 3rem;
    line-height: 3.8rem;
  }

  .c0 .CoC {
    font-size: 8rem;
    line-height: 6.5rem;
  }

  .c0 .definition-container {
    font-size: 6.5rem;
    line-height: 5.5rem;
    margin-bottom: 300px;
  }

  .c0 .definition {
    font-size: 6rem;
  }

  .c0 .OR {
    font-size: 4.5rem;
    width: 150px;
    height: 150px;
    margin-left: 120px;
    margin-right: 120px;
  }

  .c0 .button {
    width: 600px;
    height: 150px;
    border: solid black 4px;
    font-size: 5rem;
  }
}

<div
  className="c0 primary-background"
>
  <header
    className="c1 header"
  >
    <div
      className="header-text"
    >
      <p
        className="welcome-to"
      >
        WELCOME TO
      </p>
      <h1
        className="CoC"
      >
        CARDS OF CAROUSAL
      </h1>
    </div>
  </header>
  <main>
    <div
      className="definition-container"
    >
      <p
        className="word"
      >
        CAROUSAL 
        <span
          className="definition"
        >
          (n)
        </span>
      </p>
      <p
        className="definition"
      >
        a wild, drunken party or celebration : a drunken revel
      </p>
    </div>
    <div
      className="button-container"
    >
      <button
        className="c2 button"
        onClick={[Function]}
        type="button"
      >
        <p>
          JOIN
        </p>
      </button>
      <div
        className="OR"
      >
        OR
      </div>
      <button
        className="c2 button host-button"
        onClick={[Function]}
        type="button"
      >
        <p>
          HOST
        </p>
      </button>
    </div>
  </main>
  <footer
    className="c3"
  >
    <p>
      Card content thanks to:
       
      <a
        href="https://cardsagainsthumanity.com"
      >
        https://cardsagainsthumanity.com
      </a>
    </p>
    <p>
      Made by Odin students, with love:
       
      <a
        href="https://www.theodinproject.com"
      >
        https://www.theodinproject.com
      </a>
    </p>
  </footer>
</div>
`;
